tests/support/dict_repo.py

```diff
@@ -0,0 +1,102 @@
+        # Check if the entity already exists in the repo
+        identifier = getattr(entity, entity.id_field)
+        if self.db[self.schema.name].get(identifier):
+            raise DuplicateObjectError(
+                f'Entity with id {identifier} already exists')
+
+        # Add the entity to the repository
+        self.db[self.schema.name][identifier] = self.schema.from_entity(entity)
+        return self.db[self.schema.name][identifier]
+
+    def _read(self, page=1, per_page=10, order_by=(), **filters):
+        """ Read the repository and return results as per the filer"""
+
+        # Filter the dictionary objects based on the filters
+        items = []
+        for item in self.db[self.schema.name].values():
+            for fk, fv in filters.items():
+                if item[fk] != fv:
+                    break
+            else:
+                items.append(item)
+
+        # Sort the filtered results based on the order_by clause
+        for o_key in order_by:
+            reverse = False
+            if o_key.startswith('-'):
+                reverse = True
+                o_key = o_key[1:]
+            items = sorted(items, key=itemgetter(o_key), reverse=reverse)
+
+        # Build the pagination results for the filtered items
+        cur_offset = (page - 1) * per_page
+        cur_limit = page * per_page
+        result = Pagination(
+            page=page,
+            per_page=per_page,
+            total=len(items),
+            items=items[cur_offset: cur_limit])
+        return result
+
+    def _update(self, entity: Entity):
+        """ Update the entity record in the dictionary """
+        identifier = getattr(entity, entity.id_field)
+        self.db[self.schema.name][identifier] = self.schema.from_entity(entity)
+        return self.db[self.schema.name][identifier]
+
+    def delete(self, identifier):
+        """ Delete the dictionary object by its id"""
+
+        # Delete the object from the dictionary and return the deletion count
+        del_count = 0
+        try:
+            del self.db[self.schema.name][identifier]
+            del_count += 1
+        except KeyError:
+            pass
+        return del_count
+
+
+class DictSchema(RepositorySchema):
+
+    class Meta:
+        entity = Entity
+
+    def from_entity(self, entity):
+        """ Convert the entity to a dictionary record """
+        dict_obj = {}
+        for field_name in entity.declared_fields:
+            dict_obj[field_name] = getattr(entity, field_name)
+        return dict_obj
+
+    def to_entity(self, item):
+        """ Convert the dictionary record to an entity """
+        return self.opts.entity(item)
+
+
+class DictRepositoryFactory(RepositoryFactory):
+
+    def get_connection(self):
+        """ Return the dictionary database object """
+        if not self._connection:
+            self._connection = defaultdict(dict)
+        return self._connection
+
+
+drf = DictRepositoryFactory()
```

Comment:
Eventually, `DictRepositoryFactory` should be a singleton that accepts an entity class and supplies the prepared model/schema/repo (already connected to the database). Correct?

Reply:
Yes it is meant to be a singleton for registering a schema with a repository and can return the repository for a schema